clusterloader2/docs/experiments.md

```diff
@@ -0,0 +1,97 @@
+breakages aren't that costly (unless they happen on the weekend, see
+https://github.com/kubernetes/perf-tests/pull/586), they are expensive for
+large, rarely run tests (e.g. [ci-kubernetes-e2e-gce-scale-performance]).
+
+For this reason, all new changes/features added to clusterloader2 shall be gated
+and rolled out gradually. This way, we can minimize blast radius of breaking
+changes. We can even stop some kinds of them to happen at all, as they should be
+caught at presubmit time and not allowed to merge at all.
+
+## Experiment rollout
+
+### Experiment design
+
+
+### General principles
+
+1. Allow at least 24h between changes to test configs to ensure the experiment
+   is stable. We don't want to block PR to k/k because of a flaky experiment in
+   clusterloader2.
+
+1. Check with sig-scalability whether there is a regression in the test you want
+   to enable experiment for. We don't want new features in clusterloader2 to
+   interfere with regression' debugging.
+
+1. For tests not listed below it's fine to enable experiments at your
+   convenience.
+
+### Step-by-step process
+
+_Each step should be a separate PR_
+
+1. Add a "knob" to turn on the future experiment. Usually this means adding a
+   new environmental variable (PR to [test-infra]) or a new override file (PR to
+   [perf-tests]). At this point the "knob" is not used by any code path.
+
+   Since the knob is not used anywhere it's no-op and should be safe.
+
+
+1. Enable experiment in [perf-tests presubmits] and
+   [ci-kubernetes-e2e-gci-gce-scalability]
+
+   Again cince the knob is not used anywhere it's no-op and should be safe.
+
+   Perf-test presubmit runs two jobs: [pull-perf-tests-clusterloader2] and
+   [pull-perf-tests-clusterloader2-kubemark]. Primary role of those presubmits
+   is to catch bugs in code from perf-tests, so we enable the experiment for
+   both of those jobs first. Enabling experiment in
+   [ci-kubernetes-e2e-gci-gce-scalability] should give you enough data points to
+   determine whether experiments work, once we add new code path or config. It
+   also run frequent enough, so in case of problems you can revert quickly.
+   Before enabling the experiment in [ci-kubernetes-e2e-gci-gce-scalability]
+   make sure there is no ongoing regression affecting this test.
+
+1. Add a new code path or config that uses "knob" added in the first PR.
+
+   We've already enable it in the first step, so the PR will be only merged if
+   the new code path or configuration passes [perf-tests presubmits].
+
+1. Enable experiment in [pull-kubernetes-e2e-gce-100-performance] and
+   [pull-kubernetes-kubemark-e2e-gce-big]
+
+   Changing presubmit definitions in test-infra has an ability to break k/k
+   presubmits. PRs to test-infra don't trigger presubmit in k/k. Once you enable
+   the experiment in the presubmit you need to watch next 3 runs after your PR
+   is merged to detect breakages.
+
+1. Enable experiment in [ci-kubernetes-e2e-gce-scale-performance]
```

Comment:
And all other kubemarks?

Reply:
No strong opinion on this. @mm4tt?

Reply:
Not sure if I follow, I don't see any step modifying kubemark tests (except presubmits) and I think it should be listed here (especially if we recommend changing kubemark presubmit)

Reply:
I've re-added Kubemark test as part of the last step.